priority_queue.py

```diff
@@ -0,0 +1,47 @@
+from queue import Queue
+
+
+class Priority_Queue(object):
+
+    def __init__(self):
+        self.high = Queue()
+        self.low = Queue()
+
+    def insert(self, item, priority="high"):
+        """
+        Insert a queue first by priority, then by first-in-last-out.
+
+        Priority_Queue does not track the absolute order that all items
+        were inserted into the queue, and only provides Priority_Queue
+        behavior through pop() and insert() functions.
+        Priority levels are abritrarily set to only 'high' or 'low'.
+        """
+        if priority == "high":
+            self.high.enqueue(item)
+        elif priority == "low":
+            self.low.enqueue(item)
+        else:
+            raise ValueError("Priorty must be literal 'high' or 'low'.")
+
+    def pop(self):
+        """
+        Dequeue higher priority objects first, then those of lower priority.
+
+        If the Priority_Queue is empty, the Queue underneath will raise a
+        IndexError.
+        """
+        if self.high.size():
```

Comment:
this is asking permission rather than forgiveness.  Don't do it.  Just dequeue the empty queue, catch the error, and move on.